instrumentation/grpc-1.6/library/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/TracingClientInterceptor.java

```diff
@@ -55,8 +56,13 @@
         throw e;
       }
     }
-
-    SocketAddress address = result.getAttributes().get(Grpc.TRANSPORT_ATTR_REMOTE_ADDR);
+    SocketAddress address = null;
+    try {
+      URI uri = new URI(null, next.authority(), null, null, null);
+      address = new InetSocketAddress(uri.getHost(), uri.getPort() == -1 ? 80 : uri.getPort());
```

Comment:
Won't calling the `InetSocketAddress` constructor cause DNS resolution here? Should we use `InetSocketAddress.createUnresolved()` instead?
On the other hand - it's a client instrumentation, so in theory the address should already be resolved - is that the case here? Or is the intercept method called earlier than that?

Reply:
> On the other hand - it's a client instrumentation, so in theory the address should already be resolved - is that the case here? Or is the intercept method called earlier than that?

Yes, I found that invoked `result.getAttributes().get(Grpc.TRANSPORT_ATTR_REMOTE_ADDR)` will return `null`. But in `TracingClientCallListener#onMessage` method, invoked it will get the expected result. So the root cause should be `result.getAttributes()` called too early.

Reply:
@ralphgj I think the concern here is that calling [new InetSocketAddress](https://docs.oracle.com/javase/8/docs/api/java/net/InetSocketAddress.html#InetSocketAddress-java.lang.String-int-) will trigger the host name to be resolved to an IP address, which we try to avoid triggering from instrumentation

`InetSocketAddress.createUnresolved()` is probably safer unless we are clear this won't be a problem

Reply:
Got it. I tried and then found the attributes in spans without `net.peer.ip`. Shall we go on using `InetSocketAddress.createUnresolved()` instead?

Reply:
ya, it's ok not to capture `net.peer.ip`